cli/cmd/logs.go

```diff
@@ -0,0 +1,195 @@
+	clientset *kubernetes.Clientset
+	*stern.Config
+}
+
+type logFlags struct {
+	containerFilter string
+	namespace       string
+	podFilter       string
+	containerState  string
+	labelSelector   string
+	sinceSeconds    time.Duration
+	tail            int64
+	timestamps      bool
+	follow          bool
+}
+
+func (c *logFlags) NewSternConfig() (*stern.Config, error) {
+	config := &stern.Config{}
+
+	podFilterRgx, err := regexp.Compile(c.podFilter)
+	if err != nil {
+		return nil, err
+	}
+	config.PodQuery = podFilterRgx
+
+	containerFilterRgx, err := regexp.Compile(c.containerFilter)
+	if err != nil {
+		return nil, err
+	}
+	config.ContainerQuery = containerFilterRgx
+
+	containerState, err := stern.NewContainerState(c.containerState)
+	if err != nil {
+		return nil, err
+	}
+	config.ContainerState = containerState
+
+	funcs := make(map[string]interface{})
+
+	funcs["color"] = func(color color.Color, text string) string {
+		return color.SprintFunc()(text)
+	}
+
+	tmpl, err := template.New("logs").
+		Funcs(funcs).Parse("{{color .PodColor .PodName}} {{color .ContainerColor .ContainerName}} {{.Message}}")
+	if err != nil {
+		return nil, err
+	}
+	config.Template = tmpl
+
+	if c.labelSelector == "" {
+		config.LabelSelector = labels.Everything()
+	} else {
+		selector, err := labels.Parse(c.labelSelector)
+		if err != nil {
+			return nil, err
+		}
+		config.LabelSelector = selector
+	}
+
+	if c.tail != -1 {
+		config.TailLines = &c.tail
+	}
+
+	config.Since = c.sinceSeconds
+	config.Timestamps = c.timestamps
+	config.Namespace = c.namespace
+
+	return config, nil
+}
+
+func newLogOptions(flags *logFlags, kubeconfigPath, kubeContext string) (*logCmdOpts, error) {
+	kubeAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext)
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(kubeAPI.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := flags.NewSternConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return &logCmdOpts{
+		clientset,
+		c,
+	}, nil
+}
+
+func newCmdLogs() *cobra.Command {
+
+	flags := &logFlags{}
+
+	cmd := &cobra.Command{
+		Use:   "logs [flags]",
+		Short: "Tail logs from kubernetes resource",
+		Long:  `Tail logs from kubernetes resource.`,
+		Example: `  # Tail logs from all containers in pods that have the prefix 'linkerd-controller' in the linkerd namespace
+  linkerd logs --pod-filter linkerd-controller.* --namespace linkerd
+
+  # Tail logs from the linkerd-proxy container in the grafana pod within the linkerd control plane
+  linkerd logs --pod-filter linkerd-grafana.* --container linkerd-proxy --namespace linkerd
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts, err := newLogOptions(flags, kubeconfigPath, kubeContext)
+
+			if err != nil {
+				return err
+			}
+
+			return runLogOutput(opts)
+		},
+	}
+
+	cmd.PersistentFlags().StringVarP(&flags.containerFilter, "container", "c", "", "Regex string to use for filtering log lines by container name")
```

Comment:
This flag is great because it effectively lets you select the control plane component that you care about.  e.g. `linkerd logs -c public-api`.  

We could also consider changing this to not accept a regex and also updating the doc string to say that this must be one of "public-api", "private-api", "tap", "destination", "prometheus", "grafana", or "linkerd-proxy".  This could help with discovery.  This would also allow us to add tab completion as a future improvement.